### PR TITLE
New module: execute_lambda (AWS)

### PR DESCRIPTION
**Pull request by ryansb**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

`execute_lambda`
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel c4cff768d9) last updated 2016/07/12 14:36:47 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 799159b8ee) last updated 2016/07/12 14:36:21 (GMT -400)
  lib/ansible/modules/extras: (ryansb-execute-lambda 02b850694d) last updated 2016/07/13 13:56:40 (GMT -400)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

New module for the AWS Lambda [invoke API](https://docs.aws.amazon.com/lambda/latest/dg/API_Invoke.html) - with this module you can invoke a function from Ansible with any arguments, and use its output, logs, and status in future tasks. 

This module is not idempotent because the act of invocation is, by definition, not idempotent and something that will happen on every playbook run. 

Example task:

```
  - name: async invoke
    execute_lambda:
      name: "hello-world"
      payload:
        key1: foo
        key2: bar
        key3: baz
  - name: sync invoke
    execute_lambda:
      name: "hello-world"
      await: yes
      payload:
        key1: foop
        key2: bar
        key3: baz
  - name: sync invoke w/ logs
    execute_lambda:
      name: "hello-world"
      tail_log: yes
      payload:
        key1: foop
        key2: bar
        key3: baz
  - name: test/dryrun invoke
    execute_lambda:
      name: "hello-world"
      dry_run: yes
```

Output:

```
PLAYBOOK: 2016-07-lambda-exec.yml **********************************************
1 plays in 2016-07-lambda-exec.yml

PLAY [localhost] ***************************************************************

TASK [async invoke] ************************************************************
task path: /home/ryansb/code/ansible-test-plays/2016-07-lambda-exec.yml:5
changed: [localhost] => {"changed": true, "result": {"logs": "", "output": "foo", "status": 200}}

TASK [sync invoke] *************************************************************
task path: /home/ryansb/code/ansible-test-plays/2016-07-lambda-exec.yml:14
changed: [localhost] => {"changed": true, "result": {"logs": "", "output": "foop", "status": 200}}

TASK [sync invoke w/ logs] *****************************************************
task path: /home/ryansb/code/ansible-test-plays/2016-07-lambda-exec.yml:24
changed: [localhost] => {"changed": true, "result": {"logs": "START RequestId: a2aae369-4923-11e6-a912-a9b105b24ded Version: $LATEST\nvalue1 = fo$p\nvalue2 = bar\nvalue3 = baz\nEND RequestId: a2aae369-4923-11e6-a912-a9b105b24ded\nREPORT RequestId: a2aae369-4923-11e6-a912-a9b105b24ded\tDurat$on: 0.34 ms\tBilled Duration: 100 ms \tMemory Size: 128 MB\tMax Memory Used: 21 MB\t\n", "output": "foop", "status": 200}}

TASK [test/dryrun invoke] ******************************************************
task path: /home/ryansb/code/ansible-test-plays/2016-07-lambda-exec.yml:34
changed: [localhost] => {"changed": true, "result": {"logs": "", "output": "", "status": 204}}
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @ryansb for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/execute_lambda.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        function_arn         = dict(),
+        await                = dict(choices=BOOLEANS, default=True, type='bool'),
+        tail_log             = dict(choices=BOOLEANS, default=False, type='bool'),
+        dry_run              = dict(choices=BOOLEANS, default=False, type='bool'),
+        version_qualifier    = dict(),
+        payload              = dict(default={}, type='dict'),
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
```
Comment (mattclay) [inline cloud/amazon/execute_lambda.py:153]: The module sets `supports_check_mode=True`, but there is no logic to handle check mode. Shouldn't check mode be used to enable a dry run?

Comment (ryansb) [inline cloud/amazon/execute_lambda.py:153]: Ah, oops - I meant to have that in there. I changed the DryRun logic to execute `if dry_run or module.check_mode`.

**Comment by shawnsi** [comment]
@ryansb thanks for this module!  I ran a few test cases and it all works as expected.

shipit

**Comment by ryansb** [comment]
Improved exception traceback handling and switch `await` to `wait`.

**Comment by juliedavila** [comment]
shipit

**Pull request merged**